src/classes/RD2_RecurringDonation.cls

```diff
@@ -154,6 +154,64 @@ public inherited sharing class RD2_RecurringDonation {
         return this;
     }
 
+    /**
+    * @description Revises the schedules' EndDate of fixed length recurring donation
+    * to control how many fixed length future installments should be created
+    * @return RD2_RecurringDonation This record instance
+    */
+    public RD2_RecurringDonation reviseScheduleEndDateForFixedLength(RD2_ScheduleService scheduleService) {
+        if (!isFixedLength() || isFixedLengthComplete()) {
+            return this;
+        }
+
+        List<RD2_OpportunityMatcher.Record> futureInstallments = getFutureInstallments(scheduleService);
+
+        Date scheduleEndDate = (futureInstallments.isEmpty())
+            ? RD2_ScheduleService.currentDate.addDays(-1)
+            : futureInstallments[futureInstallments.size() - 1].getCloseDate();
+
+        for (RecurringDonationSchedule__c schedule : rd.RecurringDonationSchedules__r) {
+            if (schedule.EndDate__c == null) {
+                schedule.EndDate__c = scheduleEndDate;
+            }
+        }
+
+        return this;
+    }
+
+    /**
+    * @description Get matching records of all future scheduled installments and opened opportunities for the given fixed length Recurring Donation
+    * @return List<RD2_OpportunityMatcher.Record>
+    */
+    private List<RD2_OpportunityMatcher.Record> getFutureInstallments(RD2_ScheduleService scheduleService) {
+        List<RD2_OpportunityMatcher.Record> installments = new List<RD2_OpportunityMatcher.Record>();
+        Integer existingInstallmentsCount = getPaidInstallments() + getOpenOpportunities().size();
+
+        if (!isFixedLength() || isFixedLengthComplete() || (existingInstallmentsCount >= getPlannedInstallments())) {
```

Comment:
The latest [commit](https://github.com/SalesforceFoundation/NPSP/pull/5212/commits/78c4824cc206e9872468c9bae12e16302c6bc528) includes these changes. What is the scenario/test that cause the change?